documentation/List-of-Action-plugins.md

```diff
@@ -406,6 +408,73 @@ The list of changed review/responses submitted is passed as `changedResponses` t
 
 ---
 
+### Generate Document
+
+Generates a PDF file based on a [Carbone](https://carbone.io/api-reference.html) document template.
+
+- _Action Code:_ **`generateDoc`**
+
+| Input parameters<br />(\*required) <br/> | Output properties                          |
+| ---------------------------------------- | ------------------------------------------ |
+| `docTemplateId`\*                        | `document: {uniqueId, filename, filepath}` |
+| `data`                                   |                                            |
+| `additionalData`                         |                                            |
+| `userId`                                 |                                            |
+| `applicationSerial`                      |                                            |
+
+The Action utilises the internal `generatePDF` function, which is also accessible via the [`/generate-pdf` endpoint](API.md)
+
+`docTemplateId` specifies the uniqueId of the carbone template file (from the "file" table)
+
+`data` is the object that contains all the data for substituting values into the carbone template. You can specifify a `data` object explicitly, or by default it will use properties `applicationData` and `outputCumulative`. If you want to _add_ to these, you can provide an `additionalData` parameter object which will be added as an additional field, like so:
+
+```
+data: { ...applicationData, ...outputCumulative, additionalData }
+```
+
+`userId` and `applicationSerial` are not required for functioning, but will be stored as fields in the resulting "file" record. If `applicationSerial` is supplied, the output PDF file will be stored in a subfolder named for the application serial.
```

Comment:
If you store a file, it's normally useful to associate a user with it (though not essential). If you upload a file using the file upload plugin, the userId gets stored.

I don't think we do anything with it yet, but it will likely be useful at some point to see all the files associated with a given user.